core/domain/stats_jobs_continuous.py

```diff
@@ -526,41 +526,118 @@ def entity_classes_to_map_over(cls):
+        # versions.
+        if exploration_version != VERSION_ALL and len(versions) != 1:
+            yield (
+                'Expected a single version when aggregating answers for '
+                'exploration %s (v=%s), but found: %s' % (
+                    exploration_id, exploration_version, versions))
+
+        # Map interaction IDs and StateAnswersModel IDs to exploration versions.
+        versioned_interaction_ids = {version: set() for version in versions}
+        versioned_item_ids = {version: set() for version in versions}
+        for value_dict in value_dicts:
+            version = value_dict['exploration_version']
+            versioned_interaction_ids[version].update(
+                [value_dict['interaction_id']])
+            versioned_item_ids[version].update([value_dict['item_id']])
+
+        # Convert the interaction IDs to a list so they may be easily indexed.
+        versioned_interaction_ids = {
+            v: list(interaction_ids)
+            for v, interaction_ids in versioned_interaction_ids.iteritems()
+        }
+
+        # Verify all interaction ID and item ID containers are well-structured.
+        for version, interaction_ids in versioned_interaction_ids.iteritems():
+            if len(interaction_ids) != 1:
+                yield (
+                    'Expected exactly one interaction ID for exploration %s '
+                    'and version %s, found: %s' % (
+                        exploration_id, version, len(interaction_ids)))
+        for version, item_ids in versioned_item_ids.iteritems():
+            if not item_ids:
+                yield (
+                    'Expected at least one item ID for exploration %s and '
+                    'version %s, found: %s' % (
+                        exploration_id, version, len(item_ids)))
+
+        # Filter out any item IDs which happen at and before a version with a
+        # changed interaction ID. Start with the most recent version since it
+        # will refer to the most relevant answers.
+        latest_version = versions[0]
+        latest_interaction_id = versioned_interaction_ids[latest_version][0]
+        # NOTE TO DEVELOPER: This skips the first version in the list since that
+        # interaction ID is already being compraed. Bear in mind that skipping
+        # the first version shifts all found indexes back by one, which is
+        # intended here since the index found is the first index whose
+        # interaction ID has changed, which means the version before that in the
+        # list is desired. If all interaction IDs are the same, then the last
+        # version is used. All of these operations no-op when versions is a list
+        # of 1 version, which happens for each individually mapped version (see
+        # the map function). For this reason, all of this code is simply skipped
+        # in that case in favor of performance.
+        if len(versions) > 1:
+            start_version_index = next(
```

Comment:
Convoluted. Strongly recommend breaking down into multiple statements.

"start_version" doesn't really convey what you want IMO. Maybe earliest_version_with_current_interaction_id

Or actually -- earliest_acceptable_version may be nicer. (Shorter, and still conveys more than "start".)